pkg/multus/multus.go

```diff
@@ -744,65 +865,62 @@ func CmdAdd(args *skel.CmdArgs, exec invoke.Exec, kubeClient *k8s.ClientInfo) (c
-	if n.Kubeconfig != "" && kc != nil {
-		if !types.CheckSystemNamespaces(string(k8sArgs.K8S_POD_NAME), n.SystemNamespaces) {
-			err = k8s.SetNetworkStatus(kubeClient, k8sArgs, netStatus, n)
-			if err != nil {
-				if strings.Contains(err.Error(), "failed to query the pod") {
-					return nil, cmdErr(k8sArgs, "error setting the networks status, pod was already deleted: %v", err)
+	if cniConf.Kubeconfig != "" && kc != nil {
+		if !types.CheckSystemNamespaces(mc.PodName, cniConf.SystemNamespaces) {
+			if err := k8s.SetPodNetworkStatusAnnotation(kubeclient, mc.PodName, mc.PodNamespace, mc.PodUID, netStatus, cniConf); err != nil {
+				if strings.Contains(err.Error(), "failed to query the GetPod") {
+					return nil, cmdErr(mc.printableK8sArgs(), "error setting the networks status, GetPod was already deleted: %v", err)
 				}
-				return nil, cmdErr(k8sArgs, "error setting the networks status: %v", err)
+				return nil, cmdErr(mc.printableK8sArgs(), "error setting the networks status: %v", err)
 			}
 		}
 	}
 
 	return result, nil
 }
 
-//CmdCheck ...
-func CmdCheck(args *skel.CmdArgs, exec invoke.Exec, kubeClient *k8s.ClientInfo) error {
-	in, err := types.LoadNetConf(args.StdinData)
-	logging.Debugf("CmdCheck: %v, %v, %v", args, exec, kubeClient)
-	if err != nil {
-		return err
-	}
-
-	k8sArgs, err := k8s.GetK8sArgs(args)
-	if err != nil {
-		return cmdErr(nil, "error getting k8s args: %v", err)
+func (mc *CNICmd) printableK8sArgs() *types.K8sArgs {
+	return &types.K8sArgs{
+		K8S_POD_NAME:      cnitypes.UnmarshallableString(mc.PodName),
+		K8S_POD_NAMESPACE: cnitypes.UnmarshallableString(mc.PodNamespace),
+		K8S_POD_UID:       cnitypes.UnmarshallableString(mc.PodUID),
 	}
+}
 
-	for idx, delegate := range in.Delegates {
-		ifName := getIfname(delegate, args.IfName, idx)
-
-		rt, _ := types.CreateCNIRuntimeConf(args, k8sArgs, ifName, in.RuntimeConfig, delegate)
-		err = delegateCheck(exec, ifName, delegate, rt, in)
-		if err != nil {
+// CmdCheck is the handler of the CNI CHECK command
+func (mc *CNICmd) CmdCheck(cniConf *types.NetConf, exec invoke.Exec) error {
+	for idx, delegate := range cniConf.Delegates {
+		ifName := getIfname(delegate, mc.InterfaceName, idx)
+
+		rt, _ := types.CNIRuntimeConf(
+			mc.ContainerID,
+			mc.SandboxID,
+			mc.PodName,
+			mc.PodNamespace,
+			mc.PodUID,
+			mc.NetnsPath,
+			ifName,
+			cniConf.RuntimeConfig,
+			delegate)
+		if err := delegateCheck(exec, ifName, delegate, rt, cniConf); err != nil {
 			return err
 		}
 	}
-
 	return nil
 }
 
-//CmdDel ...
-func CmdDel(args *skel.CmdArgs, exec invoke.Exec, kubeClient *k8s.ClientInfo) error {
-	in, err := types.LoadNetConf(args.StdinData)
-	logging.Debugf("CmdDel: %v, %v, %v", args, exec, kubeClient)
-	if err != nil {
-		return err
-	}
-
+// CmdDelete is the handler of the CNI DEL command
+func (mc *CNICmd) CmdDelete(cniConf *types.NetConf, pod *v1.Pod, exec invoke.Exec, kubeclient *k8s.ClientInfo) error {
```

Comment:
To me, `mc *CNICmd` seems to be part of args, not method variable.
We could keep the function name as it is. Is there any clear reason why we need to change the function name?

Reply:
same as in https://github.com/k8snetworkplumbingwg/multus-cni/pull/771#discussion_r782252596

I can preserve the function name if you prefer.